modules/lang-painless/build.gradle

```diff
@@ -49,17 +52,41 @@ task apiJavadoc(type: Javadoc) {
   include '**/org/elasticsearch/painless/api/'
   destinationDir = new File(docsDir, 'apiJavadoc')
 }
+
 task apiJavadocJar(type: Jar) {
   classifier = 'apiJavadoc'
   from apiJavadoc
 }
+
 assemble.dependsOn apiJavadocJar
 
-// Reference documentation for Painless's public API.
-task generatePainlessApi(type: JavaExec) {
-  main = 'org.elasticsearch.painless.PainlessDocGenerator'
-  classpath = sourceSets.test.runtimeClasspath
-  args file('../../docs/painless/painless-api-reference')
+/**********************************************
+ *           Context API Generation           *
+ **********************************************/
+
+sourceSets {
+  doc {
+    java {
+      srcDirs = ['src/doc/java']
+    }
+  }
+}
+
+dependencies {
+  docCompile "org.elasticsearch:elasticsearch:${project.versions.elasticsearch}"
+  docCompile project(':modules:lang-painless')
+}
+
+ClusterConfiguration clusterConfig = project.extensions.create("generateContextCluster", ClusterConfiguration.class, project)
```

Comment:
@atorok we should probably avoid using `ClusterFormationTasks` and instead use the `TestClustersPlugin` here if that's possible. Would that work here?

Reply:
We tried using testclusters, but it there are issues with it here. I don't think it is ready for general use yet.

Reply:
We are using it for all plugins  and modules now, so i'm +1 on adding new tests with testclusers. Wouldn't determinedly be interested in hearing why it didn't work.

Reply:
I agree with @atorok we are full swing migrating at this point. We should have very good reason for _adding_ new usages of `ClusterFormationTasks`. The only valid reason I could see is if test clusters does not yet support a specific capability required by these tests.

Reply:
IIRC, testclusters didn't work before because painless is a module, and even though we were not using it for integTest, the check in PluginBuildPlugin would throw an error since the testclusters plugin was applied in a module. Since it is now used by modules, we may be able to get it working.

Reply:
Yeah. I think that is supported now. @atorok can confirm.

Reply:
Testclusters still does not work here. I think there needs to be more buildSrc tests for standalone uses, as I see a lot of initialization and wiring done inside PluginBuildPlugin. The configuration we tried was:

```
testClusters {
  docExtractor {
    version = project.version
    javaHome = project.file(project.ext.runtimeJavaHome)
    distribution = "DEFAULT"
  }
}

task generateContextDoc(type: JavaExec) {
  main = 'org.elasticsearch.painless.ContextDocGenerator'
  classpath = sourceSets.doc.runtimeClasspath
  useCluster testClusters.docExtractor
  systemProperty "cluster.uri", "${-> testClusters.docsExtractor.httpSocketURI}"
}
```

In addition to the extra settings of version and java home that should be defaulted to reasonable values and not require setting explicitly,  it seems to fail to depend on actually building the distribution:
> Caused by: org.elasticsearch.gradle.testclusters.TestClustersException: Can not start node{:modules:lang-painless:docExtractor-1}, missing: /mnt/ramdisk/elasticsearch/build/testclusters/extract/org.elasticsearch.distribution.elasticsearch/elasticsearch-8.0.0-SNAPSHOT

I don't think this PR should be blocked on making test clusters work for non integ tests.